src/session/client-certificate.c

```diff
@@ -102,6 +117,40 @@ read_proc_self_cgroup (size_t *out_length)
   return result;
 }
 
+static
+unsigned long long get_proc_pid_start_time (int dirfd)
+{
+  int fd = openat (dirfd, "stat", O_RDONLY);
+  if (fd < 0)
+    err (EX, "Failed to open /proc/pid/stat");
+
+  char buffer[4096];
+  ssize_t size = read (fd, buffer, sizeof buffer);
+  close (fd);
+  if (size < 0)
+    err (EX, "Failed to read /proc/pid/stat");
+
+  /* start time is the token at index 19 after the '(process name)' entry - since only this
+  * field can contain the ')' character, search backwards for this to avoid malicious
+  * processes trying to fool us; See proc_pid_stat(5) */
+  char *p = strrchr (buffer, ')');
```

Comment:
You didn't nul-terminate `buffer` after the read.  I'm a bit surprised that this didn't crash.

...well, I guess it's likely to hit a `nul` at some point.  The stack grows down, after all....

Reply:
See above, the code doesn't expect to ever actually hit the end of the buffer. But you are right, it should null terminate it just in case.

Reply:
Whoops! Fixed.